dht/token.hh

```diff
@@ -82,21 +93,25 @@ public:
     }
 
     size_t external_memory_usage() const {
-        return _data.external_memory_usage();
+        return 0;
     }
 
     size_t memory_usage() const {
-        return sizeof(token) + external_memory_usage();
+        return sizeof(token) + sizeof(_data);
     }
 
     explicit token(token_view v) : _kind(v._kind), _data(v._data) {}
 
     operator token_view() const {
         return token_view(*this);
     }
+
+    bytes_view data() const {
+        return bytes_view(reinterpret_cast<int8_t*>(const_cast<uint8_t*>(_data.begin())), 8);
```

Comment:
Is this casting an array iterator to a pointer?

Reply:
Array iterators are a pointer.

But why is the const_cast needed? you can reinterpret_cast to `const int8_t*`, that's all bytes_view needs.

You can use _data.size() instead of 8, nicer not to hard-code assumptions (even though we know they cannot change).

Reply:
> Array iterators are a pointer.

The C++ standard says that the iterator type is implementation-defined. I don't think we should depend on it (especially if we want to compile Scylla with compilers other than g++ in the future).

In any case, this is an abstraction leak.

Reply:
I did bytes_view(reinterpret_cast<const int8_t*>(_data.data()), _data.size()) which is better. Don't worry about this too much though - it goes away in the following patches.

Reply:
@kbr- I was sure that std::array's iterators are always pointers, but can't find evidence.